app/addons/dataimporter/stores.js

```diff
@@ -0,0 +1,369 @@
+// Licensed under the Apache License, Version 2.0 (the "License"); you may not
+// use this file except in compliance with the License. You may obtain a copy of
+// the License at
+//
+//   http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+// WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+// License for the specific language governing permissions and limitations under
+// the License.
+
+define([
+  'app',
+  'api',
+  'addons/dataimporter/actiontypes',
+  'papaparse'
+], function (app, FauxtonAPI, ActionTypes, Papa) {
+
+  Papa.SCRIPT_PATH = '../../assets/js/libs/papaparse.min.js';
+
+  var DataImporterStore = FauxtonAPI.Store.extend({
+
+    init: function (firstTimeHere) { //to reset, call this with true
+      if (firstTimeHere) {
+        this._isDataCurrentlyLoading = false;
+        this._hasDataLoaded = false;
+        this._hasErrored = false;
+        this._theData = [];
+        this._theMetadata = [];
+        this._smallData = [];
+        this._showView = 'table';
+        this._theFile = { size: 0 };
+        this._config = this.getDefaultConfig();
+        this._completeFn = this._config.complete;
+        this._errorFn = this._config.error;
+        this._fileSize = 0;
+        this._time = "just started";
+        this._repeatTimeID;
+        this.fetchAllDBs();
+        this._chunkedData = [];
+        this._maxSize = 150000000;  //in bytes
+        this._showErrorScreen = false;
+        this._errorMessageArray = [];
+        this._loadingInDBInProgress = false;
+      } // else keeps store as it was when you left the page
+      this.fetchAllDBs();
+
+    },
+
+    fetchAllDBs: function () {
+      $.ajax({
+        method: "GET",
+        dataType: "json",
+        url: window.location.origin + "/_all_dbs"
+      }).then(function (resp) {
+        this._all_dbs = _.filter(resp, function (dbName) {
+          return dbName[0] !== '_'; //_all_dbs without _ as first letter
+        });
+        this.triggerChange();
+      }.bind(this));
+    },
+
+    getMaxSize: function () {
+      return this._maxSize;
+    },
+
+    getAllDBs: function () {
+      return this._all_dbs;
+    },
+
+    isDataCurrentlyLoading: function () {
+      return this._isDataCurrentlyLoading;
+    },
+
+    dataIsLoading: function () {
+      this._startTime = app.helpers.moment();
+      this._isDataCurrentlyLoading = true;
+    },
+
+    hasDataLoaded: function () {
+      return this._hasDataLoaded;
+    },
+
+    dataLoaded: function () {
+      this._hasDataLoaded = true;
+      clearInterval(this._repeatTimeID);
+    },
+
+    loadData: function (data) {
+      this._theData = data;
+    },
+
+    getFileSize: function () {
+      return this._theFile.size;
+    },
+
+    isThisABigFile: function () {
+      return this._theFile.size > 750000 ? true : false;
```

Comment:
I don't much like all the hardcoded values like this, 60000, 3000 and others 'cause I just don't know what they mean. I'd favour having a constants file for stuff like this just so they're in one place and can have meaningful names.